### DISCUSSION TITLE
lsp is crashing while running on cloud

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-theia/theia-cloud#162] [Q&A]
Hi, 

I am building an app based on Theia that is deployed on Kubernetes, but during the initialization of workspace, my app keeps running into the following problem

![image](https://user-images.githubusercontent.com/116638890/233033782-8cea0518-4b46-42fa-87a9-b841e58f1a96.png)

The docker image for this app works fine on local machines but is unable to run on Kubernetes.

`message: '[Error - 9:36:17 AM] Apr 19, 2023, 9:36:17 AM Cannot run program "kill": error=2, No such file or directory\n' +
    'Cannot run program "kill": error=2, No such file or directory\n' +
    'java.io.IOException: Cannot run program "kill": error=2, No such file or directory\n' +
    '\tat java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)\n' +
    '\tat java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)\n' +
    '\tat java.base/java.lang.Runtime.exec(Runtime.java:594)\n' +
    '\tat java.base/java.lang.Runtime.exec(Runtime.java:418)\n' +
    '\tat java.base/java.lang.Runtime.exec(Runtime.java:315)\n' +
    '\tat org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.parentProcessStillRunning(ParentProcessWatcher.java:77)\n' +
    '\tat org.eclipse.jdt.ls.core.internal.ParentProcessWatcher.run(ParentProcessWatcher.java:49)\n' +
    '\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n' +
    '\tat java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)\n' +
    '\tat java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n' +
    '\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n' +
    '\tat java.base/java.lang.Thread.run(Thread.java:833)\n' +
    'Caused by: java.io.IOException: error=2, No such file or directory\n' +
    '\tat java.base/java.lang.ProcessImpl.forkAndExec(Native Method)\n' +
    '\tat java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)\n' +
    '\tat java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)\n' +
    '\tat java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)\n' +
    '\t... 12 more\n',
  level: 'info',
  timestamp: '2023-04-19 09:36:17.559'
}`

**Comment by jfaltermeier** [comment]
Hi, from the pasted Log I can't tell what is going on. 
I would suggest to check where the language server is creating its workspace directory by running the docker image on a local machine via `ps -aux`, e.g.:

```
/usr/lib/jvm/temurin-17-jdk-amd64/bin/java --add-modules=ALL-SYSTEM --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED -Declipse.application=org.eclipse.jdt.ls.core.id1 -Dosgi.bundles.defaultStartLevel=4 -Declipse.product=org.eclipse.jdt.ls.core.product -Dfile.encoding=utf8 -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xmx1G -Xms100m -jar /tmp/.mount_TheiaB2qc9eo/resources/app/plugins/redhat.java/extension/server/plugins/org.eclipse.equinox.launcher_1.6.0.v20200915-1508.jar -configuration /home/user/.theia/globalStorage/redhat.java/0.73.0/config_linux -data /home/user/.theia/workspace-storage/4ab60c63984633177b0b1165ed0e7b47/redhat.java/jdt_ws
```

The `-data` option points to the workspace. Inside the workspace directory, there should be log file (`/home/user/.theia/workspace-storage/4ab60c63984633177b0b1165ed0e7b47/redhat.java/jdt_ws/.metadata/.log`). 

Then check the log file at a similar location when running it on kubernetes.
